lib/fluent/plugin/buf_file.rb

```diff
@@ -195,8 +195,23 @@ def generate_chunk(metadata)
       end
 
       def handle_broken_files(path, mode, e)
-        log.error "found broken chunk file during resume. Deleted corresponding files:", :path => path, :mode => mode, :err_msg => e.message
-        # After support 'backup_dir' feature, these files are moved to backup_dir instead of unlink.
+        log.error "found broken chunk file during resume.", :path => path, :mode => mode, :err_msg => e.message
+        unique_id = Fluent::Plugin::Buffer::FileChunk.unique_id_from_path(path)
+        if @disable_chunk_backup
+          log.warn "disable_chunk_backup is true. #{dump_unique_id_hex(unique_id)} chunk is thrown away"
+          return
+        end
```

Comment:
It might be better to move into `backup` method as same as `Fluent::Output::backup_chunk`, so that:
* `@disable_chunk_backup` is only referred in lib/fluent/buffer.rb, it makes easy to read.
* If other child classes are added, they can use same implementation.

Reply:
Indeed! This logic should be in the parent `buffer` class. I will fix it.

Reply:
I just remembered why I didn't do that.

The function of throwing away the chunks should be in the child class, so it was difficult to move this warning message to the parent class.

But I think I can do something good about this. I will try.

Reply:
I will fix this based on the following idea.

The parent class `Buffer` should have the logic not to back up with `@disable_chunk_backup` setting.

On the other hand, the logic of throwing chunks is for child classes, so the warning message should remain in them.

Reply:
I have fixed this @ 6152b8fc111adb1e137546ab6c1bc5958e3682bc
I separated the warning message into the parent class and the child classes.

```
 [info]: #0 starting fluentd worker pid=773861 ppid=773841 worker=0
 [error]: #0 [test_id] found broken chunk file during resume. path="/test/fluentd/buffer/buffer.b5f48ea187091b2e4a46d3b1fa7e0a730.log" mode=:staged err_msg="staged meta file is broken. no implicit conversion of Symbol into Integer"
 [warn]: #0 [test_id] disable_chunk_backup is true. 5f48ea187091b2e4a46d3b1fa7e0a730 chunk is not backed up.
 [warn]: #0 [test_id] disable_chunk_backup is true. 5f48ea187091b2e4a46d3b1fa7e0a730 chunk is thrown away.
 [info]: #0 fluentd worker is now running worker=0
```